src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -70,6 +77,42 @@
                               ('$(RunAnalyzers)' == '' and '$(RunAnalyzersDuringBuild)' == 'false')">
       <_SkipAnalyzers>true</_SkipAnalyzers>
     </PropertyGroup>
+
+    <!-- PERF: For builds which are indirectly triggered inside Visual Studio from commands 
+               such as from 'Run Tests' or 'Start Debugging', we implicitly skip analyzers to speed up the build.
+               We only do so by default when 'TreatWarningsAsErrors' is not 'true'. For the scenario where
+			         'TreatWarningsAsErrors' is 'true', users can explicitly enable this functionality by setting
+			         'OptimizeIndirectlyTriggeredBuildInsideVisualStudio' to 'true'.
+    -->
+    <PropertyGroup Condition="'$(_SkipAnalyzers)' == '' and
+                              '$(IsIndirectlyTriggeredBuildInsideVisualStudio)' == 'true' and
+                              ('$(TreatWarningsAsErrors)' != 'true' or '$(OptimizeIndirectlyTriggeredBuildInsideVisualStudio)' == 'true')">
```

Comment:
Consider that editor config can also cause warnings to be promoted to errors. That is another hidden way that warnings can be promoted. Should we be suppressing additional files when we are skipping analyzers? They shouldn't really have a purpose in that mode correct?

Reply:
> Consider that editor config can also cause warnings to be promoted to errors. That is another hidden way that warnings can be promoted.

Yes, that is a known design decision as we lose majority of perf gains associated with the feature if we end up loading analyzers and then identifying if any of them report an error in each file and conditionally enable/disable them. We believe majority of users would retain analyzer diagnostics as warnings for their local development with warnaserror in CI to catch them. Additionally, amongst such users percentage of users who would oppose us unblocking debugging/running tests in presence of such analyzer errors would also be pretty small, if any. So for v1 we are going with the current design of only disabling the feature by default for TreatWarningsAsErrors, and iterate on it based on customer feedback.

> Should we be suppressing additional files when we are skipping analyzers? They shouldn't really have a purpose in that mode correct?

Possibly, but I am unsure if that can have side effects. For example, code fixes for compiler warnings can potentially rely on additional files.

Reply:
>Should we be suppressing additional files when we are skipping analyzers? They shouldn't really have a purpose in that mode correct?

They are still used by generators and so required for correctness of the build.

Reply:
🤦‍♂️ yep, feel silly for not thinking of that.